custom_components/frigate/api.py

```diff
@@ -112,14 +112,25 @@ async def async_get_event_summary(
     async def async_get_config(self) -> dict[str, Any]:
         """Get data from the API."""
         return cast(
-            Dict[str, Any],
+            dict[str, Any],
             await self.api_wrapper("get", str(URL(self._host) / "api/config")),
         )
 
     async def async_get_path(self, path: str) -> Any:
         """Get data from the API."""
         return await self.api_wrapper("get", str(URL(self._host) / f"{path}/"))
 
+    async def async_retain(self, event_id: str, retain: bool) -> dict[str, Any]:
+        """Un/Retain an event."""
+        return cast(
+            dict[str, Any],
+            await self.api_wrapper(
+                "post" if retain else "delete",
+                str(URL(self._host) / f"api/events/{event_id}/retain"),
+                decode_json=True,
+            ),
+        )
+
     async def api_wrapper(
```

Comment:
I don't see the json `data` being passed in here / being accessed. I know it probably isn't used right now but I'd say either it shouldn't be a parameter or the fun should handle the case it is not None. Perhaps I am missing something though.

Reply:
I'm referring to the data parameter `data: dict | None = None`

Reply:
Oh! Yeah, that's an issue. As you say, it's just that we're not POSTing anything at the moment which is why everything still works, but I've fixed that now for future proofing. Take a look and make sure that's what you meant?

Reply:
Yep that's exactly what I was meaning 👍